src/zimscraperlib/types.py

```diff
@@ -40,7 +40,7 @@ def get_mime_for_name(
     filename: str | pathlib.Path,
     fallback: str | None = FALLBACK_MIME,
     no_ext_to=ARTICLE_MIME,
-) -> str:
+) -> str | None:
```

Comment:
This is not a API change, this is an update on what the method was/is really returning. Method code did not changed. It is already covered by `Many changes in type hints to match the real underlying code`. It is not expecting to break any scraper.

Reply:
I think that accepting `None` as `fallback` and `no_ext_to` is mandatory because in same cases we might want to not fallback in these situations to do whatever is needed when mimetype is not detected, so I prefer to still allow `None` for these situations.

Reply:
> This is not a API change, this is an update on what the method was/is really returning

Signature changed, it's an API change. The fact that the signature was fixed and not the code is irrelevant. From user's POV, previous behavior could have been a bug. I think it's good practice to report this in changelog (even if in this case it has zero impact).

>  so I prefer to still allow None for these situations.

I meant  we could even fallback with empty str or None. We should be consistent and use None though. 👍